http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -0,0 +1,316 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package http
+package service
+
+import java.time.Instant
+import java.util.UUID
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.Parallel
+import cats.effect.concurrent.Semaphore
+import cats.effect.{Async, Blocker, ContextShift, Timer}
+import cats.implicits._
+import cats.mtl.ApplicativeAsk
+import com.google.auth.oauth2.AccessToken
+import com.google.cloud.BaseServiceException
+import io.chrisdavenport.log4cats.StructuredLogger
+import org.broadinstitute.dsde.workbench.google2.util.RetryPredicates
+import org.broadinstitute.dsde.workbench.google2.{GcsBlobName, GoogleStorageService}
+import org.broadinstitute.dsde.workbench.leonardo.RuntimeImageType.{Jupyter, Welder}
+import org.broadinstitute.dsde.workbench.leonardo.config.{AutoFreezeConfig, DataprocConfig, GceConfig, ImageConfig}
+import org.broadinstitute.dsde.workbench.leonardo.dao.DockerDAO
+import org.broadinstitute.dsde.workbench.leonardo.db.{DbReference, SaveCluster, clusterQuery}
+import org.broadinstitute.dsde.workbench.leonardo.http.api.{CreateRuntime2Request, RuntimeServiceContext}
+import org.broadinstitute.dsde.workbench.leonardo.http.service.LeonardoService._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.RuntimeServiceInterp._
+import org.broadinstitute.dsde.workbench.leonardo.model._
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage.CreateCluster
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo, WorkbenchEmail, google}
+
+import scala.concurrent.ExecutionContext
+
+class RuntimeServiceInterp[F[_]: Parallel](blocker: Blocker,
```

Comment:
So the intent is to keep `LeonardoService` for the legacy routes, and build up `RuntimeService` / `RuntimeServiceInterp` as we go for the new routes, right? That's probably fine, but I hope we can have clients switch over and delete `LeoRoutes` and `LeonardoService` soon to avoid too much code duplication. Also we should add a comment to `LeonardoService`/`LeoRoutes` explaining the situation.